src/main.ts

```diff
@@ -10,6 +11,9 @@ async function bootstrap(): Promise<void> {
   const applicationPort: string =
     configurationService.getOrThrow('application.port');
 
+  const databaseMigrator = app.get(DatabaseMigrator);
+  await databaseMigrator.migrate();
```

Comment:
This ties into my comment regarding the locking approach. I think this would make sense "localised" in an `onModuleInit` within the migrator service. It can then be hidden behind a feature flag easier, for example. What do you think?

Reply:
I was considering this approach, but I personally prefer being explicit, especially since we want the migrations to execute on startup. This way, it’s clearer what runs at startup rather than hiding it in `onModuleInit`.

Reply:
Whilst I tend to agree, as we are feature-flag reliant and certain database-reliants features are behind some, e.g. accounts, this would prevent startup without enabling the database (which currently isn't the case on prod.)